couchbase/json-queries.yml

```diff
@@ -0,0 +1,151 @@
+type: json
+data:
+  url: /admin/vitals
+  server: ${COUCHBASE_QUERY_HOSTPORT}
+  basicHttpAuthUsername: ${COUCHBASE_BASICAUTH_USERNAME}
+  basicHttpAuthPassword: ${COUCHBASE_BASICAUTH_PASSWORD}
+
+observation:
+  - name: queries
+    metricNamespace: couchbase
+    path: $.
+
+    metric:
+      - name: cores
+        source: cores
+        type: long_gauge
+        label: cores
+        description: Cores
+
+      - name: cpu.sys.percent
+        source: cpu.sys.percent
+        type: long_gauge
+        label: system cpu
+        unit: "%"
+        description: Percentage of CPU used by the system
+
+      - name: cpu.user.percent
+        source: cpu.user.percent
+        type: long_gauge
+        label: user cpu
+        unit: "%"
+        description: Percentage of CPU used by the user
+
+      - name: gc.num
```

Comment:
Because...?

Reply:
Couchbase has N different components, I am not sure these gc.xxxx metrics are about JVM component which could be monitored with our agent (note that this would also means that besides JSON endpoint config would have to contain JMX endpoint). I guess Tien will know which gc metrics these are.

Reply:
Couchbase can have N nodes (may be physic nodes) for query services and we may not need to install our agent on any of those nodes. we only pull metrics using api. so our agent can't "know" automatically.